src/components/network/network-renderer.js

```diff
@@ -565,3 +571,99 @@ genotet.NetworkRenderer.prototype.findSelectEdges = function(edgeIds) {
     this.signal('showMultiEdges');
   }
 };
+
+/**
+ * Initializes the polygon selection mode.
+ * @private
+ */
+genotet.NetworkRenderer.prototype.polygonSelectionInit_ = function() {
+  var coords = [];
+  var line = d3.svg.line();
+  var tmpNodes = [];
+  for (var nodeId in this.nodes_) {
+    tmpNodes.push({
+      x: this.nodes_[nodeId].x * this.zoomScale_ + this.zoomTranslate_[0],
+      y: this.nodes_[nodeId].y * this.zoomScale_ + this.zoomTranslate_[1],
+      id: nodeId
+    });
+  }
+  var dot = this.svgNodes_.selectAll('rect, circle');
+  this.drag_ = d3.behavior.drag()
+    .on('dragstart', function() {
+      dot.classed('active', function() {
+        return false;
+      });
+      coords = [];
+      this.svgPath_.selectAll('path').remove();
+    }.bind(this))
+    .on('drag', function() {
+      coords.push(d3.mouse(this.canvas.node()));
+      this.data.selectedNodes = {};
+      tmpNodes.forEach(function(node) {
+        var point = [node.x, node.y];
+        if (genotet.vector.pointInPolygon(point, coords)) {
+          this.data.selectedNodes[node.id] = true;
+        }
+      }, this);
+      dot.classed('active', function(node) {
+        return node.id in this.data.selectedNodes;
+      }.bind(this));
+      this.drawPath_(false, coords, line);
+    }.bind(this))
+    .on('dragend', function() {
+      this.drawPath_(true, coords, line);
+    }.bind(this));
+};
+
+/**
+ * Switches the network polygon-selection/zoom mode.
+ */
+genotet.NetworkRenderer.prototype.switchMode = function() {
+  if (this.data.options.mouseZoom) {
+    this.zoomMode_();
+  } else {
+    this.polyMode_();
+  }
+};
+
+/**
+ * Transfers the network view to polygon selection mode.
+ * @private
+ */
+genotet.NetworkRenderer.prototype.polyMode_ = function() {
+  this.canvas.on('.zoom', null);
+  this.polygonSelectionInit_();
+  this.canvas.call(this.drag_);
+};
+
+/**
+ * Transfers the network view to zoom mode.
+ * @private
+ */
+genotet.NetworkRenderer.prototype.zoomMode_ = function() {
+  this.canvas.on('.drag', null);
+  this.canvas.call(this.zoom_);
+};
+
+/**
+ * Draws one edge of a polygon for polygon selection.
+ * @param {boolean} terminator Whether it is the last point for the polygon.
+ * @param {!Array<!Array<number>>} coords The coordinates for the polygon.
+ * @param {!d3.line} line The line drawing function.
+ * @private
+ */
+genotet.NetworkRenderer.prototype.drawPath_ = function(terminator, coords,
+                                                  line) {
+  this.svgPath_.append('path')
+    .attr({
+      d: line(coords)
+    });
+  if (terminator && coords.length) {
+    this.svgPath_.select('#terminator').remove();
+    this.svgPath_.append('path').attr({
+        id: 'terminator',
+        d: line([coords[0], coords[coords.length - 1]])
+      });
+    this.svgPath_.selectAll('path').remove();
```

Comment:
Is it possible that we have multiple paths to be removed? If not, using single selection could be clearer.
```js
this.svgPath_.select('path').remove();
```